### PR TITLE
obs-studio: resolve failed wrapping if a plugin has no data path

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Description of changes

fixes a regression due to #186548 where the symlinkJoin fails when a
package does not have that path such as `obs-studio-plugins.wlrobs`

Only one `symlinkJoin` is necessary

Try the following before and after this PR

```nix
{ pkgs ? import ./. {} }:
pkgs.wrapOBS {
  plugins = with pkgs.obs-studio-plugins; [ wlrobs obs-multi-rtmp ];
}
```

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by PedroHLC — APPROVED** [review]
:thinking: I don't see a good way to predict which ones don't have a said directory.

The said change was added to maybe avoid a conflict, but considering how on other distros people throw those files in the FHS, we can blindly trust it won't happen here too.

TL;DR: I'm okay with this reversion.

**Review by superherointj — APPROVED** [review]

**Comment by 06kellyjac** [comment]
@PedroHLC This isn't a reversion, it's identical in effect to what the other PR does just it doesn't make 2 separate symlink collections that hard enforce the directories exist

```
λ nix-shell ./test.nix --command run_test
-- Testing --
-- Test results --
Unwrapped modules loaded: 21
Wrapped modules loaded: 21
```

using the test script from the other PR

**Comment by PedroHLC** [comment]
> @PedroHLC This isn't a reversion, it's identical in effect to what the other PR does just it doesn't make 2 separate symlink collections that hard enforce the directories exist
> 
> ```
> λ nix-shell ./test.nix --command run_test
> -- Testing --
> -- Test results --
> Unwrapped modules loaded: 21
> Wrapped modules loaded: 21
> ```
> 
> using the test script from the other PR

I used the term "reversion" since the other PR first proposed this solution, and I asked to split it in the 2 separate symlinks.

Sorry for the confusion :sweat_smile:
